lib/split/user.rb

```diff
@@ -83,11 +83,29 @@ def alternative_key_for_experiment(experiment)
     private
 
     def keys_without_experiment(keys, experiment_key)
-      keys.reject { |k| k.match(Regexp.new("^#{experiment_key}(:finished)?$")) || k.match("#{experiment_key}:time_of_assignment") }
+      if experiment_key.include?(':')
+        sub_keys = keys.reject { |k| k == experiment_key }
+        sub_keys.reject do |k|
+          if k.match(Regexp.new("^#{experiment_key}:"))
+            sub_str = k.partition(':').last
+            sub_str.scan(Regexp.new("\\D")).any?
+          else
+            false
+          end
+        end
+      else
+        keys.select do |k|
+          k.match(Regexp.new("^#{experiment_key}:\\d+(:|$)")) ||
```

Comment:
starts with `experiment_key`, followed by all digits, then `:` or nothing behind.
for example:
`experiment_key:123`
`experiment_key:123:sth`